src/tools/pub/cmd/pub-add.md

```diff
@@ -0,0 +1,70 @@
+---
+title: dart pub add
+description: Use dart pub add to add a dependency.
+toc: false
+---
+
+_Add_ is one of the commands of the [pub tool](/tools/pub/cmd).
+
+```
+$ dart pub add <package>[:<constraint>] [options]
+```
+
+This command adds the specified package to the pubspec as a dependency,
+and then gets the dependency.
+
+For example, the following command is equivalent to
+editing `pubspec.yaml` (adding `http: ^0.12.1` under `dependencies`)
+and then calling `dart pub get`:
+
+```terminal
+$ dart pub add http:^0.12.1
```

Comment:
I would highlight `dart pub add http` as the primary example...

And then instead say that this is equivalent to adding `http: ^a.b.c` under dependencies, where `a.b.c` is the latest version of  package `http` that is compatible your other dependencies.

Reply:
Could we do an example `pubspec.yaml` and show before/after? Just an idea, maybe it's better to do it in a followup PR.

Reply:
@jp803333 I'll handle this, since it's going to be mostly about finessing the wording.

Reply:
I hesitate to show any code the user doesn't need to provide, because (in my experience) people who are skimming think they need to write that code.